galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt

```diff
@@ -26,14 +26,24 @@ package org.kopi.galite.domain
 open class Domain<T: Comparable<T>>(val length: Int? = null) {
 
     /**
-     * Override it if you want to define a constraint that the domain values ​​must meet.
+     * Allows to define the possible codes that the domain can take
+     *
+     * @param init
+     */
+    fun code(init: DomainCode<T>.() -> Unit): DomainType {
+        domainCode.init()
+        return DomainType.CODE
+    }
+
+    /**
+     * Override it if you want to define a constraint that the domain values ​​must verify.
      */
     open val check: ((value: T) -> Boolean)? = null
 
     /**
-     * Allows to define list of possible values that the domain can take
+     * Override it if you want to define a constraint that the domain values ​​must meet.
      */
-    open val code: (() -> Any)? = null
+    open val values: DomainType? = null
```

Comment:
Values is a variable with a plural name, which I understand is going to contain the values (types) that can be in a K domain and therefore its type must be a list or array of elements of type DomainType. If a domain can only have one value the variable name must be singular and I think that's what should be.

On the other hand, I didn't understand why the variable values can be zero? As long as the domain exists must have at least one value right? In this case it will be a lateinit variable

Reply:
You're right, then it will be changed to abstract